src/client/blocking.rs

```diff
@@ -55,6 +55,15 @@ impl Client {
         Client { inner: self.inner.with_headers(headers), runtime: self.runtime.clone() }
     }
 
+    /// Clones a new client with idempotency key headers.
+    ///
+    /// For short living idempotency keys, they only matter on push requests
+    pub fn with_idempotency_key(&self, idempotency_key: String) -> Client {
+        let mut client = self.clone();
```

Comment:
Similarly here, we are cloning the provided client, then calling inner.with_idempotency_key which performs an additional clone. Allowing the user to choose would avoid this case.

Reply:
Hi! Decided to do some reading to back up my intuition: a lot of rust libraries that provide APIs with `with_` will move the object. Examples:

- https://cs.github.com/hyperium/hyper/blob/055b4e7ea6bd22859c20d60776b0c8f20d27498e/src/server/conn.rs?q=with_#L905
 - https://cs.github.com/hyperium/hyper/blob/055b4e7ea6bd22859c20d60776b0c8f20d27498e/src/server/server.rs?q=with_#L165
 - https://cs.github.com/tokio-rs/tracing/blob/5f806286bdef217fefa0e6c3f47a6267f0855e2b/tracing-tower/src/lib.rs?q=with_#L80
 - https://cs.github.com/tokio-rs/tracing/blob/5f806286bdef217fefa0e6c3f47a6267f0855e2b/tracing/src/instrument.rs?q=with_#L186

Thanks for bringing up the inconsistency. I think we can, in deciding to follow that convention, also update the code in places that until now haven't been.

Reply:
Okay, great, I'll fix both up and resubmit the pull request when I get off from work later tonight.